src/content/blog/first-four-val-town-runtimes.mdx

```diff
@@ -0,0 +1,247 @@
+a terrific module from Kallyn Gowdy, which works by exposing a WebSocket
+connection between Deno and Node.js and wrapping it in
+a [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage)-style API.
+The "guest" code that wraps vals is also now [public](https://esm.town/guest.ts),
+though it's still an implementation detail and could change at any time.
+
+While this gives far better isolation between user code and system code
+than our previous vm and Worker-based strategies, it takes a little more
+time to start up, so we use a pool of pre-initialized processes to
+avoid that time penalty.
+
+This strategy also gives the ability to have more fine-grained communication
+with Deno, because it's based on WebSockets instead of a simple REST
+HTTP request cycle. We can send multiple messages to control vals while
+they're running, and receive multiple messages from them as they run.
+This is key to how we just cut 100ms from typical val runtimes - more on
+that later.
+
+Along with Runtime v4 was [Val Town v3, which included a slew of improvements](https://blog.val.town/blog/introducing-val-town-v3/),
+like static import support, JSX, and web-standard JavaScript.
+It was a vastly expanded set of JavaScript that "just worked,"
+and worked pretty well.
+
+#### What's next
+
+|                     | v1         | v2           | v3                   | v4                          |
+| ------------------- | ---------- | ------------ | -------------------- | --------------------------- |
+| Platform            | Node.js    | Node.js      | Deno                 | Deno                        |
+| Module support      | None       | None         | Dynamic imports only | Static + dynamic ES Modules |
+| Isolation primitive | V8 Context | V8 Context   | V8 Context           | Process                     |
+| Val syntax          | Custom JS  | Custom TS+JS | Custom TS+JS         | Standard TS+JS              |
+| Importing vals      | `@` syntax | `@` syntax   | `@` syntax           | ESM-standard `import`       |
+
+As you can tell, we've been iterating fast. v4 is not the last version of
+the Val Town runtime: we still have more ground to cover on all of the dimensions that we
+care about. It's heartening to see others contribute to this discussion - Figma's
+adoption of QuickJS, after
+[using Realms was not secure enough](https://www.figma.com/blog/an-update-on-plugin-security/),
+Amazon open-sourcing [Firecracker](https://aws.amazon.com/blogs/aws/firecracker-lightweight-virtualization-for-serverless-computing/)
+one of the components of Lambda, and Deno
+sharing [details of how their Deploy product works](https://deno.com/blog/anatomy-isolate-cloud).
+
+We want to learn from everyone. Vals should run instantly, securely, and everything should just
+work.
+
+Oh, and if you read this and you have the answer and the experience and are the
+one to unlock the next-gen runtime: we're hiring: [steve@val.town](mailto:steve@val.town).
```

Comment:
This should link to our yet-to-be-posted hiring post. In theory we should wait until that's posted to post this.

Also: our call to action could also be: try val town. It's super fun and fast and flexible, like we said above!